src/main/java/mate/jdbc/service/impl/CarServiceImpl.java

```diff
@@ -0,0 +1,68 @@
+package mate.jdbc.service.impl;
+
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.stream.Collectors;
+import mate.jdbc.dao.CarDao;
+import mate.jdbc.lib.Inject;
+import mate.jdbc.lib.Service;
+import mate.jdbc.model.Car;
+import mate.jdbc.model.Driver;
+import mate.jdbc.service.CarService;
+import mate.jdbc.service.DriverService;
+
+@Service
+public class CarServiceImpl implements CarService {
+    @Inject
+    private CarDao carDao;
+    @Inject
+    private DriverService driverService;
+
+    @Override
+    public Car create(Car car) {
+        return carDao.create(car);
+    }
+
+    @Override
+    public Car get(Long id) {
+        return carDao.get(id)
+                     .orElseThrow(
+                             () -> new NoSuchElementException("Could not find car by id " + id));
+    }
+
+    @Override
+    public List<Car> getAll() {
+        return carDao.getAll();
+    }
+
+    @Override
+    public Car update(Car car) {
+        return carDao.update(car);
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        return carDao.delete(id);
+    }
+
+    @Override
+    public void addDriverToCar(Driver driver, Car car) {
+        car.getDrivers().add(driver);
+        carDao.update(car);
+    }
+
+    @Override
+    public void removeDriverFromCar(Driver driver, Car car) {
+        car.getDrivers().remove(driver);
+        carDao.update(car);
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        Driver driver = driverService.get(driverId);
+        List<Car> cars = carDao.getAll();
+        return cars.stream()
+                   .filter(car -> car.getDrivers().contains(driver))
+                   .collect(Collectors.toList());
+    }
```

Comment:
It is not a good way to search,
Better to create a SQL query in DB
```suggestion
    @Override
    public List<Car> getAllByDriver(Long driverId) {
        return carDao.getAllByDriver(driverId);
    }
```

Reply:
Done